### ISSUE TITLE
Error when provisioning an iOS device in Japanese.

### ISSUE THREAD

**Issue by bpdegnan** [keybase/client#7105]
Keybase GUI Version: 1.0.22-20170515141716+b608f0e

When provisioning a Japanese iOS device, the device name gives an error.  The name of 携帯 caused a utf-8 error, and I generally name my devices in a manner that matches the language of the device.

**Comment by jzila** [comment]
@bpdegnan: device names need to have at least 3 utf-8 characters. We're tracking an issue on our end to make that error message clearer.

**Comment by bpdegnan** [comment]
Just FYI, it seems to be more of an issue than length:  俺の携帯 also did not
work.
I then tried 出具の携帯電話.

The error was:
Unknown error: bad signature: All JSON proof characters must be in the
visible ASCII set (properly escaped UTF8 is permissible)

**Comment by jzila** [comment]
Thanks. @patrickxb: any thoughts?

**Comment by patrickxb** [comment]
Sorry, but device names can't be utf8. 

    var deviceRE = regexp.MustCompile(`^[a-zA-Z0-9][ _'a-zA-Z0-9+-]*$`)
    var badDeviceRE = regexp.MustCompile(`  |[ '+_-]$|['+_-][ ]?['+_-]`)

    var CheckDeviceName = Checker{
    	F: func(s string) bool {
    		return len(s) >= 3 && len(s) <= 64 && deviceRE.MatchString(s) && !badDeviceRE.MatchString(s)
    	},
    	Hint: "between 3 and 64 characters long; use a-Z, 0-9, space, plus, underscore, dash and apostrophe",
    }

**Comment by patrickxb** [comment]
But @bpdegnan you should have seen a much better error than what you saw...I'll add a ticket to improve it.

**Comment by bpdegnan** [comment]
I agree that a more useful error would have been helpful.   Particularly
because the error that I saw was not what was shown in the Javascript.

I would like to comment that those text restrictions will be quite annoying
for us polyglots. I think that you will find that the French with their 5
accent marks, and basically the remainder of the non-english speaking world
will limit the user base with those restrictions.  Those annoying marks to
English speakers are rather significant to the rest of us.
